Canvas path svg parse

livecodeian opened the pull request.

livecodeian committed:
[[ Canvas ]] Implement parsing path commands as SVG pathdata.

livecodeian committed:
[[ Foundation ]] Refactor Number parsing from native string and add partial string parsing function.

livecodeian committed:
Merge remote-tracking branch 'origin/develop' into canvas_path_svg_parse

Conflicts:
	engine/src/module-canvas.cpp

runrevmark commented on engine/src/module-canvas.cpp:
```diff
@@ -2908,6 +2962,184 @@ void MCCanvasPathMakeEmpty(MCCanvasPathRef &r_path)
+		}
+			
+		default:
+			MCAssert(false);
```
I suggest using:

```
default:
  MCUnreachable();
  break;
```

runrevmark commented:
I think we should be throwing a suitable error if parsing an SVG path fails - i.e. be strict if the user does not give a valid SVG path it should say.

livecodeian committed:
[[ libgraphics ]] Handle Skia path close verb when iterating.
[[ Canvas ]] Fix some svg path data parsing issues - relative moveto command not relative. close command not setting the current point for relative commands.

livecodeian committed:
[[ Canvas ]] Remove old path instructions code. Added MCGPathGetSVGData function to retrieve path instructions as SVG format data.

livecodeian committed:
[[ Canvas ]] Add com.livecode.canvas.SVGPathParseError.

runrevmark commented on engine/src/module-canvas.cpp:
```diff
@@ -5642,6 +5669,8 @@ void MCCanvasErrorsFinalize()
 	MCValueRelease(kMCCanvasGradientStopOrderErrorTypeInfo);
 	MCValueRelease(kMCCanvasGradientTypeErrorTypeInfo);
+
+	MCValueRelease(kMCSVGPathParseErrorTypeInfo);
```
This should be in the Canvas namespace - i.e. kMCCanvasSVGPathParseErrorTypeInfo.

livecodeian committed:
[[ Canvas ]] Rename kMCSVGPathParseErrorTypeInfo to kMCCanvasSVGPathParseErrorTypeInfo

livecodeian committed:
Merge branch 'develop' of https://github.com/runrev/livecode into canvas_path_svg_parse

Conflicts:
	engine/src/module-canvas.cpp
	engine/src/module-canvas.h

livecodeian committed:
Merge branch 'develop' of https://github.com/runrev/livecode into canvas_path_svg_parse

Conflicts:
	engine/src/module-canvas.cpp

runrevmark commented:
Reviewed-by: @runrevmark.
